### PR TITLE
Use is_partial in checker and update tests.

### PR DESCRIPTION
**Pull request by reiterl**
What is meant with `empty`?
- Fields with null/None for sure
- Lists with []: I think, too, but ?
- Strings with "": I think, this is not empty!?

### CODE REVIEW

**Comment by jsangmeister** [comment]
> What is meant with `empty`?
> 
> * Fields with null/None for sure
yes
> * Lists with []: I think, too, but ?
yes
> * Strings with "": I think, this is not empty!?
yes, if not required

You should be able to remove the "non-partial" mode altogether from the checker, I think (or is it then used anywhere? I think all data is partial then). (I did not look at the code yet)

**Comment by reiterl** [comment]
I have remove the is_partial option. Is_partial is the new standard. Updated the checker.

File: openslides_backend/models/checker.py [hunk 1]
```diff
@@ -169,14 +168,9 @@ def __init__(
             in data, because they exist in same database.
         all: All collections are valid and has to be in the data
 
-        is_partial=True disables the check, that *all* collections have to be
-        explicitly given, so a non existing (=empty) collection will not raise
-        an error. Additionally, missing fields (=None) are ok, if they are not
-        required nor have a default (so required fields or fields with defaults
-        must be present).
+        Not all collections must be given and missing fields are ignore.
```
Comment (jsangmeister) [inline openslides_backend/models/checker.py:171]:
```suggestion
        Not all collections must be given and missing fields are ignored, but required fields and fields with a default value must be present.
```

File: global/data/example-data.json [hunk 2]
```diff
@@ -57,7 +56,6 @@
             "default_number": "",
             "default_structure_level": "",
```
Comment (jsangmeister) [inline global/data/example-data.json:57]: remove empty strings as well

File: openslides_backend/models/checker.py [hunk 3]
```diff
@@ -362,11 +353,7 @@ def check_normal_fields(self, model: Dict[str, Any], collection: str) -> bool:
             for field in self.models[collection]().get_fields()
             if field.required or field.default is not None
         )
-        necessary_fields = (
-            required_or_default_collection_fields
-            if self.is_partial
-            else all_collection_fields
-        )
+        necessary_fields = required_or_default_collection_fields
```
Comment (jsangmeister) [inline openslides_backend/models/checker.py:356]: unnecessary variable

**Review by jsangmeister — CHANGES_REQUESTED** [review]

File: openslides_backend/models/checker.py (hunk 1 above)
Comment (reiterl) [reply openslides_backend/models/checker.py:171]: Updated.

File: openslides_backend/models/checker.py (hunk 3 above)
Comment (reiterl) [reply openslides_backend/models/checker.py]: Removed.

File: global/data/example-data.json (hunk 2 above)
Comment (reiterl) [reply global/data/example-data.json]: Empty strings removed.

File: global/data/example-data.json [hunk 4]
```diff
@@ -4,9 +4,7 @@
         "1": {
```
Comment (jsangmeister) [inline global/data/example-data.json:4]: Some more fields that can be removed from both JSON files, if set to an empty/null value:
- all boolean variables which are set to false, if not required. (Maybe they are all required? Then it is fine)
- `meeting`: `custom_translations`, `start_time`, `end_time`
- `mediafile/pdf_information`
- `projection/options`
- empty collections in `example_data.json`: `motion_statute_paragraph`, `chat_message`
- all empty collections in `initial_data.json`

Also make sure to remove all unnecessary fields from the import tests.

**Review by jsangmeister — CHANGES_REQUESTED** [review]

**Comment by reiterl** [comment]
Cleanup of the initial data, remove most collections.

File: global/data/example-data.json (hunk 4 above)
Comment (reiterl) [reply global/data/example-data.json:4]: Done.

File: global/data/initial-data.json [hunk 5]
```diff
@@ -4,29 +4,16 @@
...
-            ],
-            "active_meeting_ids": [
-                1
-            ],
-            "archived_meeting_ids": [],
-            "template_meeting_ids": [],
-            "resource_ids": [],
-            "organization_tag_ids": [],
+            "committee_ids": [],
+            "active_meeting_ids": [],
```
Comment (jsangmeister) [inline global/data/initial-data.json:13]: You should also be able to remove these

File: global/data/initial-data.json [hunk 6]
```diff
@@ -36,50 +23,15 @@
...
-            "option_$_ids": [],
-            "vote_$_ids": [],
-            "vote_delegated_vote_$_ids": [],
-            "assignment_candidate_$_ids": [],
-            "projection_$_ids": [],
-            "vote_delegated_$_to_id": [],
-            "vote_delegations_$_from_ids": [],
-            "chat_message_$_ids": [],
-            "meeting_ids": []
+	    "meeting_ids": [],
```
Comment (jsangmeister) [inline global/data/initial-data.json:31]: same for this

File: tests/system/action/meeting/test_import.py [hunk 7]
```diff
@@ -351,10 +302,6 @@ def get_user_data(self, obj_id: int, data: Dict[str, Any] = {}) -> Dict[str, Any
             "id": obj_id,
             "password": "",
```
Comment (jsangmeister) [inline tests/system/action/meeting/test_import.py:303]: Also remove this and all other empty strings. Same goes for all the other helper methods, there are still some fields left.

**Review by jsangmeister — CHANGES_REQUESTED** [review]
Still some places left. Also remove the `needed_collections` part from `test_import.py` since they are no longer needed.

File: global/data/initial-data.json (hunk 6 above)
Comment (reiterl) [reply global/data/initial-data.json:31]: user/1 meeting_ids is needed.

File: global/data/initial-data.json (hunk 5 above)
Comment (reiterl) [reply global/data/initial-data.json:13]: Removed.

File: tests/system/action/meeting/test_import.py (hunk 7 above)
Comment (reiterl) [reply tests/system/action/meeting/test_import.py:303]: Some of the empty strings are needed, like password.

File: global/data/initial-data.json (hunk 6 above)
Comment (reiterl) [reply global/data/initial-data.json:31]: user/meeting_ids is a `NumberArrayField`. These fields are special fields with a default `default` of []. So the checker wants the `user/meeting_ids` field.

Comment (reiterl) [reply global/data/initial-data.json]: Updated this.

File: tests/system/action/meeting/test_import.py (hunk 7 above)
Comment (jsangmeister) [reply tests/system/action/meeting/test_import.py:303]: But not all of them, something like `title` or `pronoun` should not be required. Also, for e.g. password, we should also note in the models file that it is required.

**Review by jsangmeister — CHANGES_REQUESTED** [review]
See open comment. Also the `needed_collection` are still to be removed.

**Comment by reiterl** [comment]
I have removed the needed_collections part from the test_import.

**Comment by reiterl** [comment]
I removed some more empty fields from the meeting/test_import

File: tests/system/action/meeting/test_import.py (hunk 7 above)
Comment (reiterl) [reply tests/system/action/meeting/test_import.py]: Removed not needed (user) fields like `title`, `pronoun` and `password`.

Comment (reiterl) [reply tests/system/action/meeting/test_import.py]: Removed not needed fields like `title`,`pronoun` and `password`.

**Review by jsangmeister — APPROVED** [review]

**Pull request merged**